shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/profilePreferences/presentation/editAddress/components/EditAddressScreen.kt

```diff
@@ -0,0 +1,164 @@
+                onClick = {
+                    onEvent(EditAddressEvent.NavigateBack)
+                }
+            ) {
+                Icon(
+                    imageVector = Icons.Rounded.ArrowBack,
+                    contentDescription = null,
+                    tint = MaterialTheme.colorScheme.onPrimaryContainer
+                )
+            }
+        }
+        item {
+            Column(
+                modifier = Modifier.padding(horizontal = 16.dp, vertical = 8.dp),
+                verticalArrangement = Arrangement.spacedBy(16.dp)
+            ) {
+                ScreenTitle(
+                    UiText.StringRes(MR.strings.edit_address_title_text).asString(),
+                    UiText.StringRes(MR.strings.edit_address_subtitle_text).asString(),
+                    Modifier.padding(horizontal = 8.dp)
+                )
+
+                Spacer(modifier = Modifier.height(16.dp))
+
+                UserInput(
+                    state = state.streetName,
+                    placeholderText = UiText.StringRes(MR.strings.edit_address_street_name_placeholder_text)
+                        .asString(),
+                    keyboardOptions = KeyboardOptions(
+                        keyboardType = KeyboardType.Text, imeAction = ImeAction.Next
+                    )
+                ) {
+                    onEvent(EditAddressEvent.StreetNameChanged(it))
+                }
+
+                UserInput(
+                    state = state.doorNumber?.toString().orEmpty(),
+                    placeholderText = UiText.StringRes(MR.strings.edit_address_door_number_placeholder_text)
+                        .asString(),
+                    keyboardOptions = KeyboardOptions(
+                        keyboardType = KeyboardType.Number, imeAction = ImeAction.Next
+                    )
+                ) {
+                    onEvent(EditAddressEvent.DoorNumberChanged(it.toInt()))
+                }
+
+                UserInput(
+                    state = state.city,
+                    placeholderText = UiText.StringRes(MR.strings.edit_address_city_placeholder_text)
+                        .asString(),
+                    keyboardOptions = KeyboardOptions(
+                        keyboardType = KeyboardType.Text, imeAction = ImeAction.Next
+                    )
+                ) {
+                    onEvent(EditAddressEvent.CityChanged(it))
+                }
+
+                UserInput(
+                    state = state.postalCode?.toString().orEmpty(),
+                    placeholderText = UiText.StringRes(MR.strings.edit_address_postal_code_placeholder_text)
+                        .asString(),
+                    keyboardOptions = KeyboardOptions(
+                        keyboardType = KeyboardType.Number, imeAction = ImeAction.Next
+                    )
+                ) {
+                    onEvent(EditAddressEvent.PostalCodeChanged(it.toInt()))
```

Comment:
As above this will crash because of empty string cast to integer

Reply:
Done